tacs/constitutive.pyx

```diff
@@ -764,10 +764,89 @@ cdef class CompositeShellConstitutive(ShellConstitutive):
 
         self.props = [prop.getMaterialProperties() for prop in ply_list]
 
+    def generateBDFCard(self):
+        """
+        Generate pyNASTRAN card class based on current design variable values.
+
+        Returns:
+            card (pyNastran.bdf.cards.properties.shell.PCOMP): pyNastran card holding property information
+        """
+        num_plies = len(self.props)
+        cdef TACSCompositeShellConstitutive* comp_ptr = <TACSCompositeShellConstitutive*>self.cptr
+        cdef np.ndarray ply_thicknesses = np.zeros(num_plies, dtype)
+        cdef np.ndarray ply_angles = np.zeros(num_plies, dtype)
+
+        comp_ptr.getPlyThicknesses(<TacsScalar*>ply_thicknesses.data)
+        comp_ptr.getPlyAngles(<TacsScalar*>ply_angles.data)
+
+        mat_ids = []
+        for i in range(num_plies):
+            ply_id = self.props[i].getNastranID()
+            mat_ids.append(ply_id)
+
+        prop = nastran_cards.properties.shell.PCOMP(self.nastranID, mat_ids,
+                                                    ply_thicknesses.astype(float),
+                                                    np.rad2deg(ply_angles, dtype=float))
+        return prop
+
 cdef class BladeStiffenedShellConstitutive(ShellConstitutive):
     """This constitutive class models a shell stiffened with T-shaped stiffeners.
     The stiffeners are not explicitly modelled.
     Instead, their stiffness is "smeared" across the shell.
+
+    Parameters
+    ----------
+    panelPly : tacs.constitutive.OrthotropicPly object
+        Ply model to use for the panel
+    stiffenerPly : tacs.constitutive.OrthotropicPly object
+        Ply model to use for the stiffener
+    kcorr : TacsScalar
+        Shear correction factor, usually 5.0/6.0
+    panelLength : TacsScalar
+        Panel length DV value
+    panelLengthNum : int
+        Panel lenth DV number, passing a negative value tells TACS not to treat this as a DV
+    stiffenerPitch : TacsScalar
+        Stiffener pitch DV value
+    stiffenerPitchNum : int
+        DV number, passing a negative value tells TACS not to treat this as a DV
+    panelThick : TacsScalar
+        Panel thickness DV value
+    panelThickNum : int
+        DV number, passing a negative value tells TACS not to treat this as a DV
+    numPanelPlies : int
+        Number of distinct ply angles in the panel
+    panelPlyAngles : np.array of numPanelPlies TacsScalars
+        Array of ply angles in the panel
+    panelPlyFracs : np.array of numPanelPlies TacsScalars
+        Array of ply fractions in the panel
+    panelPlyFracNums : np.array of numPanelPlies int
+        Array of ply fraction DV numbers in the panel, passing negative values tells TACS not to treat that ply fraction as a DV
+    stiffenerHeight : TacsScalar
+        Stiffener height DV value
+    stiffenerHeightNum : int
+        DV number, passing a negative value tells TACS not to treat this as a DV
```

Comment:
The new docstring looks good. Could you modify the arg types to match the format of the other class docstring? This will help the hyperlinking in sphinx work correctly for each variable type. See [here](https://github.com/smdogroup/tacs/blob/master/tacs/constitutive.pyx#L731-L734) for example.

Basically changes like: `np.array of numPanelPlies TacsScalars` -> `numpy.ndarray[float or complex]` and `tacs.constitutive.OrthotropicPly object` -> `tacs.constitutive.OrthotropicPly`